src/vs/workbench/contrib/terminalContrib/accessibility/browser/textAreaSyncAddon.ts

```diff
@@ -85,14 +86,20 @@ export class TextAreaSyncAddon extends Disposable implements ITerminalAddon {
 		if (!this._terminal) {
 			return;
 		}
+		let isGuessForPrompt = false;
 		const commandCapability = this._capabilities.get(TerminalCapability.CommandDetection);
-		const currentCommand = commandCapability?.currentCommand;
+		let currentCommand = commandCapability?.currentCommand;
 		if (!currentCommand) {
 			this._logService.debug(`TextAreaSyncAddon#updateCommandAndCursor: no current command`);
 			return;
+		} else if (currentCommand.isInvalid && this._capabilities.get(TerminalCapability.CommandDetection)?.commands.length) {
```

Comment:
way to improve this? it's a partial command, so we cannot use `findIndex` and do that minus 1